Source/WebKit/UIProcess/Cocoa/UIDelegate.mm

```diff
@@ -1206,7 +1206,7 @@ static _WKAutoplayEvent toWKAutoplayEvent(WebCore::AutoplayEvent event)
     std::optional<WebCore::FrameIdentifier> mainFrameID;
     if (auto* mainFrame = frame.page() ? frame.page()->mainFrame() : nullptr)
         mainFrameID = mainFrame->frameID();
-    FrameInfoData frameInfo { frame.isMainFrame(), FrameType::Local, { }, userMediaOrigin.securityOrigin(), { }, frame.frameID(), mainFrameID, frame.process().processID() };
+    FrameInfoData frameInfo { frame.isMainFrame(), FrameType::Local, { }, userMediaOrigin.securityOrigin(), { }, frame.frameID(), mainFrameID, frame.process().processID(), frame.isMainFrame() };
```

Comment:
frame.isMainFrame() isn't correct.  I'd instead add a WebFrameProxy::isFocused() that returns true if its page is non-null and if its page's focusedFrame() is equal to it.  Then use that 3x in this PR.